src/app/plugins/animal.plugin.ts

```diff
@@ -0,0 +1,189 @@
+    if (!this._species.includes(animalLookUp)) {
+      const species = this._subspecies.find((species) => species.subSpecies.includes(animalLookUp))
+        ?.species;
+
+      searchTerms = `name=${species}&sub-species=${animalLookUp}`;
+    } else if (speciesIndex > -1) {
+      // Detects if the species has subspecies.  Then applies a random sub-species accordingly.
+      const subSpeciesList: string[] = this._subspecies[speciesIndex].subSpecies;
+      const randomSubspecies = subSpeciesList[Math.floor(Math.random() * subSpeciesList.length)];
+
+      searchTerms = `name=${animalLookUp}&sub-species=${randomSubspecies}`;
+    }
+
+    await this.container.httpService
+      .get(`${this._API_URL}species/?${searchTerms}`)
+      .then(async (response: IHttpResponse) => {
+        // Notifies the user if there was a problem contacting the server
+        if (Math.floor(response.status / 100) !== 2) {
+          message.reply(
+            `Something seems to have happened with the connection to ${this._API_URL}.`
+          );
+          return;
+        }
+
+        const [animalData] = response.data.animal;
+        await message.reply('', { files: [animalData.Image], name: 'image.jpg' });
+      })
+      .catch((err) => {
+        this.container.loggerService.warn(err);
+      });
+  }
+
+  private _getRandomAnimal(): string {
+    const animals: string[] = Array.from(this._validAnimals);
+    return animals[Math.floor(Math.random() * animals.length)];
+  }
+
+  private _makeSpeciesEmbed() {
+    if (this._speciesEmbed) {
+      return this._speciesEmbed;
+    }
+
+    this._speciesEmbed = this.container.messageService.generateEmbedList(this._species);
+    this._speciesEmbed.setColor('#0099ff').setTitle('Species');
+
+    return this._speciesEmbed;
+  }
+
+  private _makeSubspeciesEmbed(): MessageEmbed {
+    if (this._subspeciesEmbed) {
+      return this._subspeciesEmbed;
+    }
+
+    const embed = new MessageEmbed();
+    embed.setColor('#0099ff').setTitle('Subspecies');
+
+    this._subspecies.forEach((species) => {
+      embed.addField(species.species, species.subSpecies.join('\n'), true);
+    });
+
+    return (this._subspeciesEmbed = embed);
+  }
+
+  private _makeSingleSubBreedEmbed(subSpecies: string): MessageEmbed | string {
+    const subSpeciesData = this._subspecies.find((e) => e.species === subSpecies)?.subSpecies;
+
+    if (!subSpeciesData) {
+      return "This Species doesn't have any sub-species.";
+    }
+
+    const embed = new MessageEmbed();
+    embed.setColor('#0099ff').setTitle(subSpecies);
+    embed.setDescription(subSpeciesData.join('\n'));
+
+    return embed;
+  }
+
+  private _updateAnimalAPIData() {
+    // Checks if the API has passed it refresh time
+    const lastAnimalUpdate: number = Date.now() - this._LAST_UPD_TIME;
+
+    if (lastAnimalUpdate < this._ANIMAL_UPD_THRESH) {
+      return;
+    }
+
+    this._LAST_UPD_TIME = Date.now();
+
+    this.container.httpService
```

Comment:
use `await`

Reply:
^ looking at this, this is probably why that first bug is a thing thx.

Reply:
bump